### PR TITLE
📖 Improve and Simplify the Getting Started Guide

### PR DESCRIPTION
**Pull request by camilamacedo86**
- Refined the sample code implementation to keep it minimal, helping users better understand the core concepts.
- Properly implement automation to generate the sample
- Streamlined the guide by focusing on the main information and removing extraneous details.
- Ensured that the code and samples in the guide are consistent with the generated project.
- Simplified navigation by hiding the full code, making it easier for users to follow along.
- Replaced the use of the deploy image plugin with direct commands to provide more comprehensive and straightforward information.

### CODE REVIEW

**Comment by camilamacedo86** [comment]
c/c @TAM360

File: docs/book/src/getting-started/testdata/project/internal/controller/memcached_controller.go [hunk 1]
```diff
@@ -282,37 +206,12 @@ func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
-	r.Recorder.Event(cr, "Warning", "Deleting",
-		fmt.Sprintf("Custom Resource %s is being deleted from the namespace %s",
-			cr.Name,
-			cr.Namespace))
-}
-
 // deploymentForMemcached returns a Memcached Deployment object
 func (r *MemcachedReconciler) deploymentForMemcached(
 	memcached *cachev1alpha1.Memcached) (*appsv1.Deployment, error) {
-	ls := labelsForMemcached(memcached.Name)
```
Comment (camilamacedo86) [inline docs/book/src/getting-started/testdata/project/internal/controller/memcached_controller.go:308]: TODO: we will need to add 1 label due the test-controller implementation

File: docs/book/src/getting-started/testdata/project/internal/controller/memcached_controller.go [hunk 2]
```diff
@@ -321,42 +220,8 @@ func (r *MemcachedReconciler) deploymentForMemcached(
 		},
 		Spec: appsv1.DeploymentSpec{
 			Replicas: &replicas,
-			Selector: &metav1.LabelSelector{
-				MatchLabels: ls,
-			},
 			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: ls,
-				},
```
Comment (camilamacedo86) [inline docs/book/src/getting-started/testdata/project/internal/controller/memcached_controller.go:330]: TODO(camila): we will need to add 1 label due the test-controller implementation

**Review by varshaprasad96 — APPROVED** [review]
Looks good from my end. 
/lgtm
/approve

Adding in /hold, so that the PR doesn't get auto merged and @TAM360 gets a chance for review.

**Comment by varshaprasad96** [comment]
Looks like the PR needs to be rebased.

**Comment by camilamacedo86** [comment]
@TAM360 I know that you are checking the docs and seen how you can help out.
 So, if you see here (or in any place) something that can be fixed or improved 
Please, do not hesitate in raise a PR with your suggestions !!!

So, I am moving here with this one, we have many open prs and we can push improvements in follow up always !

**Pull request merged**